src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -349,6 +349,14 @@ protected override void OnHostStarted()
             _secretManager.PurgeOldFiles(Instance.ScriptConfig.RootScriptPath, Instance.TraceWriter);
         }
 
+        protected override void OnTimeout()
+        {
+            base.OnTimeout();
+            Stop();
```

Comment:
As Fabio mentions, we already get notified of a HostingEnvironment shutdown. The CancellationToken passed into ScriptHostManager.RunAndBlock() is the one that is signaled -- and we're already waiting on that to perform shutdown: [https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHostManager.cs#L116](https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHostManager.cs#L116)

I didn't like hardcoding 2000 here -- let me look at the shutdownTimeout. That sounds nicer.

---
In reply to: [75968751](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r75968751) [](ancestors = 75968751)